xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp

```diff
@@ -2670,6 +2674,58 @@ void CActiveAE::LoadSettings()
   m_settings.silenceTimeoutMinutes = settings->GetInt(CSettings::SETTING_AUDIOOUTPUT_STREAMSILENCE);
 }
 
+void CActiveAE::ValidateOutputDevices(bool saveChanges)
+{
+  std::string device = m_sink.ValidateOuputDevice(m_settings.device, false);
+
+  if (!device.empty() && device != m_settings.device)
+  {
+    CLog::LogF(LOGWARNING, "audio output device setting has been updated from '{}' to '{}'",
+               m_settings.device, device);
+
+    const AESinkDevice oldDevice = CAESinkFactory::ParseDevice(m_settings.device);
+    const AESinkDevice newDevice = CAESinkFactory::ParseDevice(device);
+
+    m_settings.device = device;
+
+    if (saveChanges && oldDevice == newDevice)
```

Comment:
The == operator doesn't follow usual expectations (equality of all members), so this condition works, but is not intuitive to read.
Maybe instead of == operator, a member function IsEquivalentTo() or IsSameDeviceAs(), or something like it.